### PR TITLE
IAM: Refactor user session management to middleware

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes https://github.com/nuts-foundation/nuts-node/issues/3005

Moves user session management to Echo middleware. Functional changes:

- If a user session already exists and preauthorized user details are provided, earlier it checked whether the user details of the subsequent "user access token flow" were the same as the one that were used to create the session (XIS sanity check). This check is now became unpractical, since the user details aren't available as-is in the session any more. Also, this is the responsibility of the calling XIS.
- Before, only 1 session cookie could exist for a user, so if the user performed a user access token flow for a different `did:web` DID, things would break (session not found/invalid). This is now fixed, by allowing multiple sessions (bound to web DID) to co-exist, allowing multi-tenancy.
- Changed the session from `Host` prefix to `Secure` prefix, as `Host` prefix does not allow a path other than `/`

### CODE REVIEW

File: auth/api/iam/usersession/util.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ */
+
+package usersession
+
+import "net/http"
+
+// CookieReader is an interface for reading cookies from an HTTP request.
+// It is implemented by echo.Context and http.Request.
+type CookieReader interface {
```
Comment (woutslakhorst) [inline auth/api/iam/usersession/util.go:25]: weird to have a util with only an interface

File: auth/api/iam/usersession/data.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	"errors"
+	"fmt"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-did/vc"
+	"time"
+)
+
+// Data is a session-bound Verifiable Credential wallet.
+type Data struct {
```
Comment (woutslakhorst) [inline auth/api/iam/usersession/data.go:31]: extremely generic name. Either name the package `session` and this struct `User` or place everything in the `iam` package and name it `usersession`?

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
I think middleware should be located next to other middleware. Maybe in http/session or http/user? This would solve most weird naming. `user.Wallet` vs `usersession.UserWallet`.... and `user.Session` vs `usersession.Data`

File: auth/api/iam/usersession/util.go (hunk 1 above)
Comment (reinkrul) [reply auth/api/iam/usersession/util.go]: Moved it all to 1 file

File: auth/api/iam/usersession/data.go (hunk 2 above)
Comment (reinkrul) [reply auth/api/iam/usersession/data.go]: moved `usersession` to `http` module, renamed package to `user` and `Data` -> `Session`

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**